File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.kt
class Solution {
fun countNicePairs(nums: IntArray): Int {
    var res = 0
    val mod = 1e9.toInt() + 7
    var map: HashMap<Int, Int> = HashMap<Int, Int>()
    for(num in nums) {
        val diff = num - num.toString().reversed().toInt()
        val count = map.getOrDefault(diff, 0)
        map.put(diff, count + 1)
        res = (res + count) % mod
    }
    return res
}
}

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.py
class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        d = {}
        for n in nums:
            diff = n-int(str(n)[::-1])
            d[diff] = d.get(diff, 0)+1
        return sum([d[x]*(d[x]-1)//2 for x in d])%(10**9+7)